Ensure that the tests find the built python module vs an install

kdt3rd opened the pull request:
This may have a slightly messy un-intended consequence of "snap-shotting" pythonpath when configured in a large python env, but eases testing confusion on systems where pythonpath is not yet set, but an (older) copy of OpenImageIO is installed.

This fixes the python tests on my home machine.

## Checklist:

<!-- Put an 'x' in the boxes as you complete the checklist items -->

- [x] I have read the [contribution guidelines](https://github.com/AcademySoftwareFoundation/OpenImageIO/blob/master/CONTRIBUTING.md).
- [ ] I have updated the documentation, if applicable.
- [x] I have ensured that the change is tested somewhere in the testsuite
  (adding new test cases if necessary).
- [ ] If I added or modified a C++ API call, I have also amended the
  corresponding Python bindings (and if altering ImageBufAlgo functions, also
  exposed the new functionality as oiiotool options).
- [x] My code follows the prevailing code style of this project. If I haven't
  already run clang-format before submitting, I definitely will look at the CI
  test that runs clang-format and fix anything that it highlights as being
  nonconforming.

kdt3rd committed:
Ensure that the tests find the built python python module vs an installed one

This may have a slightly messy un-intended consequence of
"snap-shotting" pythonpath when configured in a large python env, but
eases testing confusion on systems where pythonpath is not yet set, but
an (older) copy of OpenImageIO is installed.

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>

lgritz commented on src/cmake/testing.cmake:
```diff
@@ -110,6 +110,7 @@ macro (oiio_add_tests)
                              "OIIO_TESTSUITE_ROOT=${_testsuite}"
                              "OIIO_TESTSUITE_SRC=${_testsrcdir}"
                              "OIIO_TESTSUITE_CUR=${_testdir}"
+                             "PYTHONPATH=${CMAKE_BINARY_DIR}/lib/python/site-packages:$ENV{PYTHONPATH}"
```
Does the fact that you used `:` here account for the Windows CI failures?

kdt3rd replied:
most likely, switching to use generator expression which should be fine in cmake 3.14+ - the pathListSep config variable is not available until 3.24 or somesuch

kdt3rd committed:
Use generator expression to get correct OS separator

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>

kdt3rd committed:
Make build tree look more like install tree

Also adds python version and init.py to build tree such that windows has
a chance of working with the DLL loading

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>

kdt3rd commented:
With the latest commit, I can cleanly run the test target from the build folder (assuming I have previously downloaded the test images) without running an install, which eases the development workflow. I do not have a windows machine to test with, but windows (and ninja multiconfig) have multiple configs, so to avoid potential slicing, put the python modules in the appropriate config path - and copied the init.py in there such that it might work to load the dlls under windows (untested). Given that the CI runs the tests using the installed dist tree, I have disabled overriding the env variables in this scenario

lgritz commented on src/cmake/testing.cmake:
```diff
@@ -111,6 +111,20 @@ macro (oiio_add_tests)
+              # the github actions run all the tests from the dist tree
+              # and do an install prior to running the tests so skip
+              # setting these here
+              if (NOT DEFINED ENV{OpenImageIO_ROOT})
+                # this will ensure the appropriate fonts are found
+                set_property(TEST ${_testname} APPEND PROPERTY ENVIRONMENT
+                  "OpenImageIO_ROOT=${CMAKE_SOURCE_DIR}/src")
+              endif()
```
I'm not 100% sure I understand all the implications -- now or after future changes we aren't considering -- of setting OpenImageIO_ROOT to point to the src directory, of all places. 

And what happens if somebody is building in an environment where OpenImageIO_ROOT is already set to someplace other than this build area (maybe to an existing install)? In that case, this would not be set, and what would that mean?

Wouldn't it be more direct to set OPENIMAGEIO_FONTS to exactly the right spot? And much less likely to affect absolutely anything else?

Also, sorry about this nit, but would you mind using 4-space indentation like the surrounding code? Thanks.
